### ISSUE TITLE
Items remain in add/remove programs

### ISSUE THREAD

**Issue by keithemyers** [andrew-s-taylor/public#25]
Thank you for putting this out there!

The script removed all of the shortcuts for the Dell garbage, but the entries in add/remove programs remain, as do their entries in Apps > Installed Apps. I was watching while the script was running and "MyDell" did disappear from the list, but it returned.

I'm still able to launch Dell Remote Assist if I search for the .exe, so it appears it wasn't uninstalled.

The Debloat.log file shows that this was thrown for each one:

TerminatingError(Start-Process): "This command cannot be run due to the error: The system cannot find the file specified."
WARNING: Failed to uninstall: []

Windows 11 Enterprise 10.0.22631
Dell Vostro 14 3430

**Dell Optimizer**
AuthorizedCDFPrefix : 
Comments            : 
Contact             : 
DisplayVersion      : 3.2.1117.0
HelpLink            : 
HelpTelephone       : 
InstallDate         : 20230917
InstallLocation     : C:\Program Files\Dell\DellOptimizer\
InstallSource       : C:\Program Files\Dell\Downloaded_Installations\{A861556A-7676-4F68-8CA9-992EECE14211}\
ModifyPath          : MsiExec.exe /I{1344E072-D68B-48FF-BD2A-C1CCCC511A50}
Publisher           : Dell
Readme              : 
Size                : 
EstimatedSize       : 56230
SystemComponent     : 1
UninstallString     : MsiExec.exe /I{1344E072-D68B-48FF-BD2A-C1CCCC511A50}
URLInfoAbout        : http://www.Dell.com
URLUpdateInfo       : 
VersionMajor        : 3
VersionMinor        : 2
WindowsInstaller    : 1
Version             : 50463837
Language            : 1033
DisplayName         : Dell Optimizer
PSPath              : Microsoft.PowerShell.Core\Registry::HKEY_LOCAL_MACHINE\Software\Microsoft\Windows\CurrentVersion\Uninstall\{1344E072-D68B-48FF-BD2A-C1CCCC511A50}
PSParentPath        : Microsoft.PowerShell.Core\Registry::HKEY_LOCAL_MACHINE\Software\Microsoft\Windows\CurrentVersion\Uninstall
PSChildName         : {1344E072-D68B-48FF-BD2A-C1CCCC511A50}
PSDrive             : HKLM
PSProvider          : Microsoft.PowerShell.Core\Registry

**Dell Core Services**
AuthorizedCDFPrefix : 
Comments            : 
Contact             : 
DisplayVersion      : 1.2.359.4
HelpLink            : 
HelpTelephone       : 
InstallDate         : 20230917
InstallLocation     : C:\Program Files\Dell\
InstallSource       : C:\ProgramData\Package Cache\{F8D78AF7-4581-4751-8929-D4B63A410EBA}\
ModifyPath          : MsiExec.exe /I{4D1F6932-4892-480D-A929-98DD8E44326B}
Publisher           : Dell, Inc.
Readme              : 
Size                : 
EstimatedSize       : 196534
SystemComponent     : 1
UninstallString     : MsiExec.exe /I{4D1F6932-4892-480D-A929-98DD8E44326B}
URLInfoAbout        : https://www.dell.com
URLUpdateInfo       : 
VersionMajor        : 1
VersionMinor        : 2
WindowsInstaller    : 1
Version             : 16908647
Language            : 1033
DisplayName         : Dell Core Services
PSPath              : Microsoft.PowerShell.Core\Registry::HKEY_LOCAL_MACHINE\Software\Microsoft\Windows\CurrentVersion\Uninstall\{4D1F6932-4892-480D-A929-98DD8E44326B}
PSParentPath        : Microsoft.PowerShell.Core\Registry::HKEY_LOCAL_MACHINE\Software\Microsoft\Windows\CurrentVersion\Uninstall
PSChildName         : {4D1F6932-4892-480D-A929-98DD8E44326B}
PSDrive             : HKLM
PSProvider          : Microsoft.PowerShell.Core\Registry

**Dell SupportAssist OS Recovery Plugin for Dell Update**
AuthorizedCDFPrefix : 
Comments            : 
Contact             : 
DisplayVersion      : 5.5.7.18775
HelpLink            : 
HelpTelephone       : 
InstallDate         : 20230917
InstallLocation     : C:\Program Files\Dell\SARemediation\plugin\
InstallSource       : C:\ProgramData\Package Cache\{517FF73B-E045-4AA4-B0DD-61C65B510B2B}v5.5.7.18775\
ModifyPath          : MsiExec.exe /I{517FF73B-E045-4AA4-B0DD-61C65B510B2B}
Publisher           : Dell Inc.
Readme              : 
Size                : 
EstimatedSize       : 48448
SystemComponent     : 1
UninstallString     : MsiExec.exe /I{517FF73B-E045-4AA4-B0DD-61C65B510B2B}
URLInfoAbout        : 
URLUpdateInfo       : 
VersionMajor        : 5
VersionMinor        : 5
WindowsInstaller    : 1
Version             : 84213767
Language            : 1033
DisplayName         : Dell SupportAssist OS Recovery Plugin for Dell Update
PSPath              : Microsoft.PowerShell.Core\Registry::HKEY_LOCAL_MACHINE\Software\Microsoft\Windows\CurrentVersion\Uninstall\{517FF73B-E045-4AA4-B0DD-61C65B510B2B}
PSParentPath        : Microsoft.PowerShell.Core\Registry::HKEY_LOCAL_MACHINE\Software\Microsoft\Windows\CurrentVersion\Uninstall
PSChildName         : {517FF73B-E045-4AA4-B0DD-61C65B510B2B}
PSDrive             : HKLM
PSProvider          : Microsoft.PowerShell.Core\Registry

**MyDell Customer Connect**
AuthorizedCDFPrefix : 
Comments            : 
Contact             : Dell Technologies, Inc.
DisplayVersion      : 3.0.300.3
HelpLink            : https://www.dell.com/support
HelpTelephone       : 
InstallDate         : 20230917
InstallLocation     : C:\Program Files\Dell\MyDell\CustomerConnect\
InstallSource       : C:\ProgramData\Package Cache\{2A47DBF7-EB5C-4F0C-BE5D-C5F6AC9FF894}\
ModifyPath          : MsiExec.exe /I{61A12774-0A08-4918-828A-2B213F3766B5}
Publisher           : Dell Technologies, Inc.
Readme              : 
Size                : 
EstimatedSize       : 17786
SystemComponent     : 1
UninstallString     : MsiExec.exe /I{61A12774-0A08-4918-828A-2B213F3766B5}
URLInfoAbout        : https://www.dell.com/
URLUpdateInfo       : 
VersionMajor        : 3
VersionMinor        : 0
WindowsInstaller    : 1
Version             : 50331948
Language            : 1033
DisplayName         : MyDell Customer Connect
PSPath              : Microsoft.PowerShell.Core\Registry::HKEY_LOCAL_MACHINE\Software\Microsoft\Windows\CurrentVersion\Uninstall\{61A12774-0A08-4918-828A-2B213F3766B5}
PSParentPath        : Microsoft.PowerShell.Core\Registry::HKEY_LOCAL_MACHINE\Software\Microsoft\Windows\CurrentVersion\Uninstall
PSChildName         : {61A12774-0A08-4918-828A-2B213F3766B5}
PSDrive             : HKLM
PSProvider          : Microsoft.PowerShell.Core\Registry

**MyDell Application Management**

AuthorizedCDFPrefix : 
Comments            : 
Contact             : Dell Customer Support
DisplayVersion      : 3.0.300.3
HelpLink            : https://www.dell.com/support
HelpTelephone       : 
InstallDate         : 20230917
InstallLocation     : C:\Program Files\Dell\MyDell\ApplicationManagement\
InstallSource       : C:\ProgramData\Package Cache\{C4408F60-90BE-4625-90CE-CDB21B8F6E88}\
ModifyPath          : MsiExec.exe /I{84C4EE7F-9C1B-4061-B146-4F39EDCF4ED9}
Publisher           : Dell Technologies, Inc.
Readme              : 
Size                : 
EstimatedSize       : 32934
SystemComponent     : 1
UninstallString     : MsiExec.exe /I{84C4EE7F-9C1B-4061-B146-4F39EDCF4ED9}
URLInfoAbout        : https://www.dell.com
URLUpdateInfo       : 
VersionMajor        : 3
VersionMinor        : 0
WindowsInstaller    : 1
Version             : 50331948
Language            : 1033
DisplayName         : MyDell Application Management
PSPath              : Microsoft.PowerShell.Core\Registry::HKEY_LOCAL_MACHINE\Software\Microsoft\Windows\CurrentVersion\Uninstall\{84C4EE7F-9C1B-4061-B146-4F39EDCF4ED9}
PSParentPath        : Microsoft.PowerShell.Core\Registry::HKEY_LOCAL_MACHINE\Software\Microsoft\Windows\CurrentVersion\Uninstall
PSChildName         : {84C4EE7F-9C1B-4061-B146-4F39EDCF4ED9}
PSDrive             : HKLM
PSProvider          : Microsoft.PowerShell.Core\Registry

**Dell SupportAssist Remediation**

AuthorizedCDFPrefix : 
Comments            : 
Contact             : 
DisplayVersion      : 5.5.7.18775
HelpLink            : 
HelpTelephone       : 
InstallDate         : 20230917
InstallLocation     : C:\Program Files\Dell\SARemediation\agent\
InstallSource       : C:\ProgramData\Package Cache\{C4543FDB-3BC0-4585-B1C5-258FB7C2EA71}v5.5.7.18775\
ModifyPath          : MsiExec.exe /I{C4543FDB-3BC0-4585-B1C5-258FB7C2EA71}
Publisher           : Dell Inc.
Readme              : 
Size                : 
EstimatedSize       : 54752
SystemComponent     : 1
UninstallString     : MsiExec.exe /I{C4543FDB-3BC0-4585-B1C5-258FB7C2EA71}
URLInfoAbout        : 
URLUpdateInfo       : 
VersionMajor        : 5
VersionMinor        : 5
WindowsInstaller    : 1
Version             : 84213767
Language            : 1033
DisplayName         : Dell SupportAssist Remediation
PSPath              : Microsoft.PowerShell.Core\Registry::HKEY_LOCAL_MACHINE\Software\Microsoft\Windows\CurrentVersion\Uninstall\{C4543FDB-3BC0-4585-B1C5-258FB7C2EA71}
PSParentPath        : Microsoft.PowerShell.Core\Registry::HKEY_LOCAL_MACHINE\Software\Microsoft\Windows\CurrentVersion\Uninstall
PSChildName         : {C4543FDB-3BC0-4585-B1C5-258FB7C2EA71}
PSDrive             : HKLM
PSProvider          : Microsoft.PowerShell.Core\Registry

**Dell Update for Windows Universal**

AuthorizedCDFPrefix : 
Comments            : Scan your system, download, and apply the latest device driver and system bios updates.
Contact             : Contact Dell Technical Support
DisplayVersion      : 4.8.0
HelpLink            : https://support.dell.com
HelpTelephone       : 1-800-WWW-DELL
InstallDate         : 20230917
InstallLocation     : C:\Program Files\Dell\Update\
InstallSource       : C:\Windows\{2F5EBA96-CA6C-4419-B6C7-B371018597FF}\
ModifyPath          : MsiExec.exe /X{D52E2B1F-9241-473F-B535-27D67A286FC6}
NoModify            : 1
NoRepair            : 1
Publisher           : Dell Inc.
Readme              : C:\Program Files\Dell\Update\readme.txt
Size                : 
EstimatedSize       : 16505
UninstallString     : MsiExec.exe /X{D52E2B1F-9241-473F-B535-27D67A286FC6}
URLInfoAbout        : https://www.dell.com
URLUpdateInfo       : https://support.dell.com
VersionMajor        : 4
VersionMinor        : 8
WindowsInstaller    : 1
Version             : 67633152
Language            : 1033
DisplayName         : Dell Update for Windows Universal
PSPath              : Microsoft.PowerShell.Core\Registry::HKEY_LOCAL_MACHINE\Software\Microsoft\Windows\CurrentVersion\Uninstall\{D52E2B1F-9241-473F-B535-27D67A286FC6}
PSParentPath        : Microsoft.PowerShell.Core\Registry::HKEY_LOCAL_MACHINE\Software\Microsoft\Windows\CurrentVersion\Uninstall
PSChildName         : {D52E2B1F-9241-473F-B535-27D67A286FC6}
PSDrive             : HKLM
PSProvider          : Microsoft.PowerShell.Core\Registry

**Dell SupportAssist**

AuthorizedCDFPrefix : 
Comments            : Dell SupportAssist keeps your PC up to date and running at its best with recommended software and driver updates. It also helps resolve issues 
                      quickly by detecting common problems and sending issue details to Dell Technical Support agents on your behalf.
Contact             : 
DisplayVersion      : 3.14.2.45116
HelpLink            : https://www.dell.com/support/home/en-in/product-support/product/dell-supportassist-pcs-tablets/docs
HelpTelephone       : 
InstallDate         : 20240125
InstallLocation     : C:\Program Files\Dell\SupportAssistAgent\
InstallSource       : C:\Windows\TEMP\SupportAssistAgent\a6ee2585-411a-47be-b85a-4b221e536192\
ModifyPath          : MsiExec.exe /X{DB6164FC-CD98-471C-BD5B-5B14CAFA3186}
NoModify            : 1
NoRepair            : 1
Publisher           : Dell Inc.
Readme              : 
Size                : 
EstimatedSize       : 323072
UninstallString     : MsiExec.exe /X{DB6164FC-CD98-471C-BD5B-5B14CAFA3186}
URLInfoAbout        : http://dell.com/Support/SupportAssist
URLUpdateInfo       : 
VersionMajor        : 3
VersionMinor        : 14
WindowsInstaller    : 1
Version             : 51249154
Language            : 1033
DisplayName         : Dell SupportAssist
PSPath              : Microsoft.PowerShell.Core\Registry::HKEY_LOCAL_MACHINE\Software\Microsoft\Windows\CurrentVersion\Uninstall\{DB6164FC-CD98-471C-BD5B-5B14CAFA3186}
PSParentPath        : Microsoft.PowerShell.Core\Registry::HKEY_LOCAL_MACHINE\Software\Microsoft\Windows\CurrentVersion\Uninstall
PSChildName         : {DB6164FC-CD98-471C-BD5B-5B14CAFA3186}
PSDrive             : HKLM
PSProvider          : Microsoft.PowerShell.Core\Registry

**MyDell Components Installer**

AuthorizedCDFPrefix : 
Comments            : 
Contact             : 
DisplayVersion      : 3.0.300.3
HelpLink            : https://www.dell.com/support
HelpTelephone       : 
InstallDate         : 20230917
InstallLocation     : C:\Program Files\Dell\MyDell\Components\
InstallSource       : C:\ProgramData\Package Cache\{C40D9E1F-534E-490E-BC4B-3AFC97D51B7C}\
ModifyPath          : MsiExec.exe /I{FC24D516-8B85-4D36-8E48-0EF4E6968DFB}
Publisher           : Dell, Technologies, Inc.
Readme              : 
Size                : 
EstimatedSize       : 338946
SystemComponent     : 1
UninstallString     : MsiExec.exe /I{FC24D516-8B85-4D36-8E48-0EF4E6968DFB}
URLInfoAbout        : 
URLUpdateInfo       : 
VersionMajor        : 3
VersionMinor        : 0
WindowsInstaller    : 1
Version             : 50331948
Language            : 1033
DisplayName         : MyDell Components Installer
PSPath              : Microsoft.PowerShell.Core\Registry::HKEY_LOCAL_MACHINE\Software\Microsoft\Windows\CurrentVersion\Uninstall\{FC24D516-8B85-4D36-8E48-0EF4E6968DFB}
PSParentPath        : Microsoft.PowerShell.Core\Registry::HKEY_LOCAL_MACHINE\Software\Microsoft\Windows\CurrentVersion\Uninstall
PSChildName         : {FC24D516-8B85-4D36-8E48-0EF4E6968DFB}
PSDrive             : HKLM
PSProvider          : Microsoft.PowerShell.Core\Registry
[Debloat.log](https://github.com/andrew-s-taylor/public/files/14069516/Debloat.log)

**Comment by andrew-s-taylor** [comment]
I have just uploaded a new version which replaces the /I with /X can you try that one?

**Comment by keithemyers** [comment]
Three items remain after running the updated version:

1. Dell Optimizer
2. Dell SupportAssist OS Recovery Plugin for Dell Update
3. Dell SupportAssist Remediation

**Log snippet**

Attempting to uninstall: [Dell Optimizer]...
PS>TerminatingError(Start-Process): "This command cannot be run due to the error: The system cannot find the file specified."
WARNING: Failed to uninstall: []
Attempting to uninstall: [Dell SupportAssist OS Recovery Plugin for Dell Update]...
PS>TerminatingError(Start-Process): "This command cannot be run due to the error: The system cannot find the file specified."
WARNING: Failed to uninstall: []
Attempting to uninstall: [Dell SupportAssist Remediation]...
PS>TerminatingError(Start-Process): "This command cannot be run due to the error: The system cannot find the file specified."
WARNING: Failed to uninstall: []
Attempting to uninstall: [MyDell]...
PS>TerminatingError(Start-Process): "This command cannot be run due to the error: The system cannot find the file specified."
WARNING: Failed to uninstall: []

**Comment by andrew-s-taylor** [comment]
I've manually added those in 4.1.6, see if that works

**Comment by keithemyers** [comment]
Sorry, that didn't work either. Same error. The following script works for Dell Optimizer, so you may be able to incorporate it in your debloat script.

[https://github.com/svenriebedell/Dell-Tools-Intune-Install/blob/main/DellOptimizer_uninstall.ps1](https://github.com/svenriebedell/Dell-Tools-Intune-Install/blob/main/DellOptimizer_uninstall.ps1)

**Comment by keithemyers** [comment]
And the following uninstalled **Dell SuppportAssist Remediation** & **Dell SupportAssist OS Recovery Plugin for Dell Update**:

```
$dellSA = Get-ChildItem -Path HKLM:\SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall, HKLM:\SOFTWARE\Wow6432Node\Microsoft\Windows\CurrentVersion\Uninstall | Get-ItemProperty | Where-Object { $_.DisplayName -match "SupportAssist" } | Select-Object -Property UninstallString
 
ForEach ($sa in $dellSA) {
    If ($sa.UninstallString) {
        cmd.exe /c $sa.UninstallString /quiet /norestart
    }
}

```

**Comment by andrew-s-taylor** [comment]
I've just published 4.1.7, see if that works
